OrbitCore/LibunwindstackUnwinder.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <asm/perf_regs.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "unwindstack/MachineX86_64.h"
+#include "unwindstack/RegsX86_64.h"
+#include "unwindstack/Unwinder.h"
+
+class LibunwindstackUnwinder {
+ public:
+  bool SetMaps(const std::string& maps_buffer);
+
+  [[nodiscard]] std::vector<unwindstack::FrameData> Unwind(
+      const std::array<uint64_t, PERF_REG_X86_64_MAX>& perf_regs,
+      const char* stack_dump, uint64_t stack_dump_size);
+
+ private:
+  std::unique_ptr<unwindstack::BufferMaps> maps_{nullptr};
+
+  static constexpr size_t MAX_FRAMES = 1024;
+
+  static const std::array<int, unwindstack::X86_64_REG_LAST>
+      UNWINDSTACK_REGS_TO_PERF_REGS;
+
+  static std::string LibunwindstackErrorString(unwindstack::ErrorCode error_code) {
+    static const std::vector<const char*> ERROR_NAMES{
```

Comment:
Are these names from libunwindstack? Should this be functionality of libunwindstack? If not, I would create a "helpers" file to separate this from the core functionality.

Reply:
Yes, the names are from libunwindstack and it would be useful to have this in libunwindstack. But as this is only one private function and the use of libunwindstack should for now go through this class, I would leave this here for simplicity.